### ISSUE TITLE
Strange overextrusions surrounding features

### ISSUE THREAD

**Issue by AndrewEllis93** [supermerill/SuperSlicer#1402]
**Describe the bug**
**SS is producing strange overextruded perimeters surrounding certain features:**
![image](https://user-images.githubusercontent.com/34943186/125137768-598ffb80-e0db-11eb-9f1f-622abb3c8aa6.png)
![image](https://user-images.githubusercontent.com/34943186/125137778-5d238280-e0db-11eb-877b-5fb16a467ed3.png)
![image](https://user-images.githubusercontent.com/34943186/125137784-5f85dc80-e0db-11eb-84dd-67db1cf3f3ae.png)

**This issue does not appear to be present when printing with Prusa Slicer:**
![image](https://user-images.githubusercontent.com/34943186/125137807-6ad90800-e0db-11eb-8782-4c0378cc1375.png)

I and a few other folks in the Voron Discord have been experiencing this issue and have yet to come up with a reliable solution.

**To Reproduce**
See attached .3MF file, gcode, and config. 
I have been unable to narrow this issue down to a particular setting, and I have left many of the advanced features at defaults.
[SS-Issue.zip](https://github.com/supermerill/SuperSlicer/files/6794002/SS-Issue.zip)

**Desktop (please complete the following information):**
Windows 10 Pro
SuperSlicer version 2.3.56.6

**Comment by supermerill** [comment]
If you can reliably reproduce the artifact, can you try by removing "only one perimeter on top" ?

**Comment by AndrewEllis93** [comment]
> If you can reliably reproduce the artifact, can you try by removing "only one perimeter on top" ?

Just tried - same exact lines appear.

Someone mentioned that they were able to reduce these lines by reducing top layer flow - however that's more of a clue than a solution, as I can't lower my top layer flow without creating gaps.

I asked them to provide their photos/results on this GitHub issue.

**Comment by supermerill** [comment]
It's weird. Like the extruder stop, plastic overflowing from it. Then it moves again but the pressure isn't here yet.
Maybe too many small segments that overload the firmware/serial port ...

**Comment by AndrewEllis93** [comment]
My MCU loads appear to be okay - do you suggest I try lowering slicing resolution?

**Comment by supermerill** [comment]
You can, but i don't think it's the issue, as in your picture the problem appears in the middle of a long straight line.

**Comment by Tinchus2009** [comment]
Can this be related to my obseved problem? because it is what IM seeing in some print but much worst: #1427

**Comment by supermerill** [comment]
That may be. You can try to increase the xy decimals (to 5-6) to see if the firmware react better.

**Comment by AndrewEllis93** [comment]
While still not perfect, the best solution I found was to simply reduce perimeter accelerations to a low value. Closing

**Comment by sarchar** [comment]
![image](https://user-images.githubusercontent.com/4928176/165124224-64dab594-c3a3-4c5d-840a-c8ff7832ab99.png)

I'm having this issue and I've been fighting it for a couple months now.  I have a test print where I set to top solid layers to 0, which actually removes all internal solid layers.  The result is that the object with internal solid infill doesn't exhibit the artifact, while the one with solid infill does:

![image](https://user-images.githubusercontent.com/4928176/165124542-f9dca31a-7c82-46da-a8c4-14ee2b2a6d80.png)

I think this issue should be reopened..?

**Comment by sarchar** [comment]
I printed another test stopping the print right after the artifact layer so that I can inspect the infill. Here's what I have.

One object is printed with "0 top layers" -- it disables all the internal solid infills as well (that's weird TBH, but it does).

Gcode preview (in this image the right object has 0 top layers):
![image](https://user-images.githubusercontent.com/4928176/165224979-5c0757d4-0e47-473a-a02a-b9773f77e5a6.png)

You can clearly see the artifact on the object (sorry, they're reversed from the ones in the gcode preview):
![image](https://user-images.githubusercontent.com/4928176/165225007-d5529327-7335-41f6-8d15-2c7a657c7017.png)

The solid infill looks insanely over-extruded:

![image](https://user-images.githubusercontent.com/4928176/165225137-c0adcd2a-ae7a-4ab7-8a27-e22fa3e8aa0d.png)

It seems the artifact follows the direction the nozzle is flowing. It shows up only at the end of the extrusion. The artifact is also visible here for that reason, I just can't get a good photo of it:
![image](https://user-images.githubusercontent.com/4928176/165225255-dd62c6b2-7559-466c-b06b-4c2cd6cf29b2.png)

Some print settings: 0.2 layer height, 100% solid infill width, 140% sparse. Sparse infill 200mm/s, solid 100mm/s, top solid 60mm/s.

**Comment by sarchar** [comment]
> Because the perimeter isn't supported by a solid when there isn't (at least one) solid top layer(s), then the pressure in the nozzle can't be built up, and so can't be released (at the bad place).
> Why your printer is doing that... maybe a bit of over-extrusion (can be enforced by low pa / high accel/low jerk) is creating that, releasing the pressure at the first position where an open side allows for easier overflow.

I'm fairly certain I printed this test at various PA values and both lowered and raised accels to no effect.  I'll retest both of those variables soon.  What do you mean by "low jerk"?

**Comment by supermerill** [comment]
> What do you mean by "low jerk"?

With no/low jerk, it decelerate to a stall before accelerating in the other direction.

IMO, from the look of the picture, it seems you're over-extruding a bit, and that may be the root issue.

**Comment by MoBeltrao** [comment]
> > What do you mean by "low jerk"?
> 
> With no/low jerk, it decelerate to a stall before accelerating in the other direction.
> 
> IMO, from the look of the picture, it seems you're over-extruding a bit, and that may be the root issue.

I had exact the same problem and this post save a lot of job trying to figure out.
I was over-extruding as you said....thank you, nice tip.

**Comment by AndrewEllis93** [comment]
@sarchar @MoBeltrao @Tinchus2009 


I spent a bit of time digging into [issue 2665](https://github.com/supermerill/SuperSlicer/issues/2665).
Lo and behold, my solution for that issue also turned out to have a significant improvement for this issue too. (for me, at least).
- Documented [here](https://github.com/AndrewEllis93/Print-Tuning-Guide/blob/main/articles/troubleshooting/small_infill_areas_overextruded.md#not-connected-top-infill-superslicer).
- My comment on the other issue [here](https://github.com/supermerill/SuperSlicer/issues/2665#issuecomment-1306455603).

I'm curious if you guys are able to replicate this.

To summarize:
- Top solid: set to "not connected"
- Solid: also set to "not connected"
- Infill/perimeter overlap (encroachment): Set to 25%

The three settings below also bring some more improvement when combined with the above.
However they come with some downsides described in the linked doc above (either potential top layer scarring or excessive z-hopping on the top layers, take your pick)
- Minimum travel after retraction: 0
- Z-hop: Disabled on top layer
- eOnly retract when crossing perimeters: Enabled
